File: auth/session.js
const activeSessions = new Map();

function createSession(userId) {
  const token = `token_${userId}_${Date.now()}`;
  activeSessions.set(token, {
    userId,
    createdAt: new Date().toISOString(),
    expiresAt: new Date(Date.now() + 3600000).toISOString(), // 1 hour
  });
  return token;
}

function getSession(token) {
  return activeSessions.get(token) || null;
}

function destroySession(token) {
  return activeSessions.delete(token);
}

module.exports = { createSession, getSession, destroySession };

File: auth/session.py
import time
import secrets

_sessions = {}

def create_session(user_id):
    token = secrets.token_hex(16)
    _sessions[token] = {"user_id": user_id, "expires_at": time.time() + 3600}
    return token

def get_session(token):
    return _sessions.get(token)

def destroy_session(token):
    return _sessions.pop(token, None) is not None
